src/Microsoft.Health.Fhir.TemplateManagement/ArtifactProviders/BlobTemplateCollectionProvider.cs

```diff
@@ -97,18 +104,27 @@ public BlobTemplateCollectionProvider(BlobContainerClient blobContainerClient, I
             return templateCollection;
         }
 
-        private static async Task<List<string>> ListBlobsFlatListing(BlobContainerClient blobContainerClient, int? segmentSize, CancellationToken ct)
+        private async Task<List<string>> ListBlobsFlatListing(BlobContainerClient blobContainerClient, int? segmentSize, CancellationToken ct)
         {
             var blobs = new List<string>();
 
             var resultSegment = blobContainerClient.GetBlobsAsync(default, default, null, ct)
                 .AsPages(default, segmentSize);
 
+            long totalBlobsSize = 0;
+
             await foreach (Page<BlobItem> blobPage in resultSegment)
             {
                 foreach (BlobItem blobItem in blobPage.Values)
                 {
                     blobs.Add(blobItem.Name);
+
+                    totalBlobsSize += blobItem.Properties.ContentLength ?? 0;
+
+                    if (totalBlobsSize > _maxTemplateCollectionSizeInBytes)
+                    {
+                        throw new TemplateCollectionExceedsSizeLimitException(TemplateManagementErrorCode.BlobTemplateCollectionTooLarge, $"Total blob template collection size is larger than the size limit: {_templateCollectionConfiguration.TemplateCollectionSizeLimitMegabytes}MB.");
```

Comment:
from a user perspective I'm assuming they will see this exception if they have access to the logs. what will be the state of the container app after this exception is thrown?

Reply:
never mind - checked out the other PR. the exception is being caught and handled there

Reply:
Right now since the templates are loaded when the first request comes in, the api response will indicate this error message. 
When we change that to load templates on say app startup, then it will show in the container app logs and should fail the init of the app.